service/protocol.hpp

```diff
@@ -39,6 +39,7 @@ namespace raven
 
   //! Keywords
   static inline constexpr const char *request_keyword = "REQUEST_NAME";
+  static inline constexpr const char *request_state_keyword = "REQUEST_STATE";
```

Comment:
Nitpicky question: Do you want internal linkage here ? (i.e. one object per translation unit)
If not, you can remove the usage of `static`.
Furthermore, using `const char *` defines a pointer to a memory area containing the string, you should probably use an array type like : `const char request_state_keyword[] = "....";`